openapi/components/schemas/Subscription.yaml

```
@@ -1,10 +1,472 @@
+        type: boolean
+      endTime:
+        description: |-
+          Time and date when the trial ends.
+          If a trial is enabled on this subscription, a value must be provided.
+        type:
+          - 'string'
+          - 'null'
+        format: date-time
+  isTrialOnly:
+    description: |-
+      Specifies if a subscription ends after a trial period.
+      If this value is `true`, recurring settings are ignored.
+    type: boolean
+    default: false
+  invoiceTimeShift:
+    description: |-
+      Shifts issue time and due time of invoices for this subscription.
+
+      This setting overrides plan settings.
+      To use plan settings, set this value to `null`.
+
+      To use multiple plans in one subscription,
+      all plans must have the same billing period,
+      this property enables the customer to subscribe to different plans.
+    example: null
+    oneOf:
+      - $ref: ./InvoiceTimeShift.yaml
+      - type: 'null'
+  recurringInterval:
+    type:
+      - 'object'
+      - 'null'
+    description: |-
+      Recurring interval to override plan settings.
+      To use plan settings, set this value to `null`.
+
+      To use multiple plans in one subscription,
+      all plans must have the same recurring period length.
+    example: null
+    properties:
+      periodAnchorInstruction:
+        $ref: ./ServicePeriodAnchorInstruction.yaml
+  autopay:
+    description: |-
+      Specifies if payment attempts are made automatically.
+      If autopay is enabled, the payment is retrieved from the customer on the renewal date using the payment instrument that is set at `paymentInstrumentId`,
+      or the default payment instrument on the subscription.
+    type: boolean
+    default: true
+  startTime:
+    description: |-
+      Date and time when the subscription starts.
+      If this value is `null`, the current time is used.
+      This value cannot be more than one service period in the past.
+    type:
+      - 'string'
+      - 'null'
+    x-sortable: true
+    x-basic: true
+    example: null
+    format: date-time
+  endTime:
+    description: Date and time when the subscription ends.
+    x-sortable: true
+    type:
+      - 'string'
+      - 'null'
+    format: date-time
+    readOnly: true
+  renewalTime:
+    description: Date and time when the subscription renews.
+    type:
+      - 'string'
+      - 'null'
+    x-sortable: true
+    x-basic: true
+    format: date-time
+  rebillNumber:
+    description: Current billing period number.
+    type:
+      - 'integer'
+      - 'null'
+    readOnly: true
+    x-sortable: true
+  lineItems:
+    description: Subscription line items which queue until the next renewal (or interim) invoice is issued for the subscription.
+    readOnly: true
+    type: array
+    deprecated: true
+    items:
+      type: object
+      description: Invoice line item. Use `isInterim` property of upcoming invoice items instead.
```

suggestion: for readability
```suggestion
      description: Invoice line item. Use the `isInterim` property of upcoming invoice items instead.
```